flang-rt/lib/runtime/complex-powi.cpp

```diff
@@ -65,21 +66,23 @@ template <typename C, typename I> C tgpowi(C base, I exp) {
 #ifndef _MSC_VER
 // With most compilers, C complex is implemented as a builtin type that may have
 // specific ABI requirements
-extern "C" float _Complex RTNAME(cpowi)(float _Complex base, std::int32_t exp) {
+extern "C" RT_API_ATTRS CppTypeFor<TypeCategory::Complex, 4> RTNAME(cpowi)(
+    CppTypeFor<TypeCategory::Complex, 4> base, std::int32_t exp) {
```

Comment:
Sorry, I do not know where we stand with passing the complex values from Flang to the runtime, but the C complex was used here to match the ABI of pure C complex.  The discussion was here: https://reviews.llvm.org/D134889

This change may be unsafe, but this needs to be investigated.

Reply:
Ok. I'm a bit confused since lots of other runtime functions use the CppType and only complex-powi.cpp and complex-reduction.h use the `C` complex. I'll have a deeper look when I have more time.

Reply:
Note that the other APIs are passing args as references/pointers, and the memory layout is assumed to be the same for C and C++ complex.  The problem (or not) is passing it by value.

Reply:
Oh right! I missed that part. Thanks. I'll gonna keep a look on that.